spec/app/ics-028-cross-chain-validation/methods.md

```diff
@@ -819,16 +956,33 @@ function onChanOpenAck(
     // set the address of the distribution module account on the provider chain
     providerDistributionAccount = md.providerDistributionAccount
 
-    // initiate opening handshake for the distribution token transfer channel
-    // over the same connection as the CCV channel
-    // i.e., ChanOpenInit (as required by ICS20)
-    distributionChannelId = channelKeeper.ChanOpenInit(
-      UNORDERED, // order
-      channelKeeper.GetConnectionHops(channelIdentifier), // connectionHops: same as the CCV channel
-      "transfer", // portIdentifier
-      "transfer", // counterpartyPortIdentifier
-      "ics20-1" // version
-    )
+    if distributionChannelId == "" {
+      // initiate opening handshake for the distribution token transfer channel
+      // over the same connection as the CCV channel
+      // i.e., use handleChanOpenInit as defined in ICS-26
+      datagram = ChanOpenInit{
+          order: UNORDERED,
+          connectionHops: channelKeeper.GetConnectionHops(channelIdentifier), // same as the CCV channel
+          portIdentifier: "transfer",
+          counterpartyPortIdentifier: "transfer",
+          version: "ics20-1",
+      }
+      distributionChannelId = handleChanOpenInit(datagram)
+    }
+
+    // set the channel as the provider channel
+    providerChannel = channelIdentifier
+
+    // send pending slash requests;
+    // note: this can happen only if preCCV == false, as the ABCI application 
+    // can invoke SendSlashRequest only once the chain is upgraded to 
+    // a consumer chain, see BeginBlockInit below
+    SendPendingSlashRequests()
```

Comment:
I don't see exactly why this is necessary, it can be left to

https://github.com/cosmos/ibc/blob/01dbba90d238169c4905c7b11969ec7987e22729/spec/app/ics-028-cross-chain-validation/methods.md?plain=1#L1416

It will be later, but maybe that doesn't matter.

This is anyways far behind the current code, which only ever sends packets from the consumer in EndBlock https://github.com/cosmos/interchain-security/blob/4474a08a43feb8e8de227f6a7ee8b33f724ab9b5/x/ccv/consumer/keeper/relay.go

(Just FYI)